### PR TITLE
refactor!: remove several previously deprecated APIs

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
BREAKING CHANGE: Removed obsolete all_sessions and session properties
BREAKING CHANGE: Removed the obsolete start_activity method
BREAKING CHANGE: Removed the obsolete end_test_coverage method
BREAKING CHANGE: Removed the following obsolete arguments from the driver constructor: desired_capabilities, browser_profile, proxy
BREAKING CHANGE: Removed obsolete set_value and set_text methods
BREAKING CHANGE: Removed the obsolete MobileBy class
BREAKING CHANGE: Removed obsolete application management methods: launch_app, close_app, reset
BREAKING CHANGE: Removed obsolete IME methods: available_ime_engines, is_ime_active, activate_ime_engine, deactivate_ime_engine, active_ime_engine

### CODE REVIEW

**Review by KazuCocoa — APPROVED** [review]
Lgtm

**Review by jlipps — APPROVED** [review]

File: appium/webdriver/extensions/ime.py [hunk 1]
```diff
@@ -1,141 +0,0 @@
...
-        Android only.
-
-        deprecated:: 2.0.0
-
-        Returns:
-            :obj:`list` of :obj:`str`: The available input methods for an Android device
-        """
-        warnings.warn(
-            'The "available_ime_engines" API is deprecated and will be removed in future versions. '
-            'Use "mobile: shell" extension instead',
```
Comment (KazuCocoa) [inline appium/webdriver/extensions/ime.py:41]: Should we add `mobile` command for IME handling? Because `mobile:shell` is behind a security flag

Comment (mykola-mokhnach) [reply appium/webdriver/extensions/ime.py:41]: I did not see any people using these so far as these APIs are very specific.
Eventually they could be called anyway using adb shell extension

**Comment by easyteacher** [comment]
Why is set_value removed? How can I set a slider value now?

**Pull request merged**